src/service/user.js

```diff
@@ -0,0 +1,48 @@
+class User {
+  constructor() {
+    this.user = new Set();
+  }
+
+  create = (data) => {
+    const { user } = this;
+    const object = {
+      id: user.size + 1,
+      ...data,
+    };
+    user.add(object);
+    return object;
+  };
+
+  update = (data) => {
+    const { id, update } = data;
+    const { user } = this;
+    const updatedUser = Object.assign(
+      [...user].find((user) => user.id === Number(id)),
+      update
+    );
+    return updatedUser;
+  };
+
+  list = () => {
+    const { user } = this;
+    return [...user];
+  };
+
+  delete = (data) => {
+    const { id } = data;
+    const { user } = this;
+    user.delete([...user].find((user) => user.id === Number(id)));
```

Comment:
user.delete([...user].find((user) => user.id === Number(id))); Unnecessary use of rest operator
user.delete(user.find((user) => user.id === Number(id)));